substrate/frame/contracts/fixtures/build.rs

```diff
@@ -0,0 +1,277 @@
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Compile contracts to wasm and RISC-V binaries.
+use anyhow::Result;
+use parity_wasm::elements::{deserialize_file, serialize_to_file, Internal};
+use std::{
+	env, fs,
+	hash::Hasher,
+	path::{Path, PathBuf},
+	process::Command,
+};
+use twox_hash::XxHash32;
+
+/// Read the file at `path` and return its hash as a hex string.
+fn file_hash(path: &Path) -> String {
+	let data = fs::read(path).expect("file exists; qed");
+	let mut hasher = XxHash32::default();
+	hasher.write(&data);
+	hasher.write(include_bytes!("build.rs"));
+	let hash = hasher.finish();
+	format!("{:x}", hash)
+}
+
+/// A contract entry.
+struct Entry {
+	/// The path to the contract source file.
+	path: PathBuf,
+	/// The hash of the contract source file.
+	hash: String,
+}
+
+impl Entry {
+	/// Create a new contract entry from the given path.
+	fn new(path: PathBuf) -> Self {
+		let hash = file_hash(&path);
+		Self { path, hash }
+	}
+
+	/// Return the path to the contract source file.
+	fn path(&self) -> &str {
+		self.path.to_str().expect("path is valid unicode; qed")
+	}
+
+	/// Return the name of the contract.
+	fn name(&self) -> &str {
+		self.path
+			.file_stem()
+			.expect("file exits; qed")
+			.to_str()
+			.expect("name is valid unicode; qed")
+	}
+
+	/// Return the name of the output wasm file.
+	fn out_wasm_filename(&self) -> String {
+		format!("{}.wasm", self.name())
+	}
+}
+
+/// Collect all contract entries from the given source directory.
+/// Contracts that have already been compiled are filtered out.
+fn collect_entries(contracts_dir: &Path, out_dir: &Path) -> Vec<Entry> {
+	fs::read_dir(&contracts_dir)
+		.expect("src dir exists; qed")
+		.filter_map(|file| {
+			let path = file.expect("file exists; qed").path();
+			if path.extension().map_or(true, |ext| ext != "rs") {
+				return None;
+			}
+
+			let entry = Entry::new(path);
+			if out_dir.join(&entry.hash).exists() {
+				None
+			} else {
+				Some(entry)
+			}
+		})
+		.collect::<Vec<_>>()
+}
+
+/// Create a `Cargo.toml` to compile the given contract entries.
+fn create_cargo_toml<'a>(
+	fixtures_dir: &Path,
+	entries: impl Iterator<Item = &'a Entry>,
+	output_dir: &Path,
+) -> Result<()> {
+	let uapi_path = fixtures_dir.join("../uapi").canonicalize()?;
+	let common_path = fixtures_dir.join("./contracts/common").canonicalize()?;
+	let mut cargo_toml: toml::Value = toml::from_str(&format!(
+		"
+[package]
+name = 'contracts'
+version = '0.1.0'
+edition = '2021'
+
+# Binary targets are injected below.
+[[bin]]
+
+[dependencies]
+uapi = {{ package = 'pallet-contracts-uapi', default-features = false,  path = {uapi_path:?}}}
```

Comment:
No longer necessary.

```suggestion
uapi = {{ package = 'pallet-contracts-uapi', path = {uapi_path:?}}}
```